File: app/src/main/java/com/shuyu/apprecycler/chat/detail/dagger/ChatDetailViewOption.java
package com.shuyu.apprecycler.chat.detail.dagger;

import android.support.v7.widget.RecyclerView;

import com.shuyu.apprecycler.chat.detail.view.ChatDetailBottomView;
import com.shuyu.textutillib.EmojiLayout;
import com.shuyu.textutillib.KeyBoardLockLayout;
import com.shuyu.textutillib.RichEditText;


/**
 * view option
 * Created by guoshuyu on 2017/9/7.
 */

public class ChatDetailViewOption {

    RichEditText mChatDetailActivityEdit;
    KeyBoardLockLayout mChatDetailActivityKeyboardLayout;
    EmojiLayout mChatDetailActivitySendEmojiLayout;
    ChatDetailBottomView mChatDetailActivityBottomMenu;
    RecyclerView mRecyclerView;


    public RichEditText getChatDetailActivityEdit() {
        return mChatDetailActivityEdit;
    }

    public ChatDetailViewOption setChatDetailActivityEdit(RichEditText chatDetailActivityEdit) {
        this.mChatDetailActivityEdit = chatDetailActivityEdit;
        return this;
    }

    public KeyBoardLockLayout getChatDetailActivityKeyboardLayout() {
        return mChatDetailActivityKeyboardLayout;
    }

    public ChatDetailViewOption setChatDetailActivityKeyboardLayout(KeyBoardLockLayout chatDetailActivityKeyboardLayout) {
        this.mChatDetailActivityKeyboardLayout = chatDetailActivityKeyboardLayout;
        return this;
    }

    public EmojiLayout getChatDetailActivitySendEmojiLayout() {
        return mChatDetailActivitySendEmojiLayout;
    }

    public ChatDetailViewOption setChatDetailActivitySendEmojiLayout(EmojiLayout chatDetailActivitySendEmojiLayout) {
        this.mChatDetailActivitySendEmojiLayout = chatDetailActivitySendEmojiLayout;
        return this;
    }

    public ChatDetailBottomView getChatDetailActivityBottomMenu() {
        return mChatDetailActivityBottomMenu;
    }

    public ChatDetailViewOption setChatDetailActivityBottomMenu(ChatDetailBottomView chatDetailActivityBottomMenu) {
        this.mChatDetailActivityBottomMenu = chatDetailActivityBottomMenu;
        return this;
    }

    public RecyclerView getRecyclerView() {
        return mRecyclerView;
    }

    public void setRecyclerView(RecyclerView recyclerView) {
        this.mRecyclerView = recyclerView;
    }
}

File: app-kotlin/src/main/java/example/shuyu/recycler/kotlin/chat/detail/dagger/ChatDetailViewOption.kt
package example.shuyu.recycler.kotlin.chat.detail.dagger

import android.support.v7.widget.RecyclerView

import com.shuyu.textutillib.EmojiLayout
import com.shuyu.textutillib.KeyBoardLockLayout
import com.shuyu.textutillib.RichEditText

import example.shuyu.recycler.kotlin.chat.detail.view.ChatDetailBottomView


/**
 * view option
 * Created by guoshuyu on 2017/9/7.
 */

open class ChatDetailViewOption {

    var mChatDetailActivityEdit: RichEditText? = null
    var mChatDetailActivityKeyboardLayout: KeyBoardLockLayout? = null
    var mChatDetailActivitySendEmojiLayout: EmojiLayout? = null
    var mChatDetailActivityBottomMenu: ChatDetailBottomView? = null
    var recyclerView: RecyclerView? = null


    fun getChatDetailActivityEdit(): RichEditText {
        return mChatDetailActivityEdit!!
    }

    fun setChatDetailActivityEdit(chatDetailActivityEdit: RichEditText): ChatDetailViewOption {
        this.mChatDetailActivityEdit = chatDetailActivityEdit
        return this
    }

    fun getChatDetailActivityKeyboardLayout(): KeyBoardLockLayout {
        return mChatDetailActivityKeyboardLayout!!
    }

    fun setChatDetailActivityKeyboardLayout(chatDetailActivityKeyboardLayout: KeyBoardLockLayout): ChatDetailViewOption {
        this.mChatDetailActivityKeyboardLayout = chatDetailActivityKeyboardLayout
        return this
    }

    fun getChatDetailActivitySendEmojiLayout(): EmojiLayout {
        return mChatDetailActivitySendEmojiLayout!!
    }

    fun setChatDetailActivitySendEmojiLayout(chatDetailActivitySendEmojiLayout: EmojiLayout): ChatDetailViewOption {
        this.mChatDetailActivitySendEmojiLayout = chatDetailActivitySendEmojiLayout
        return this
    }

    fun getChatDetailActivityBottomMenu(): ChatDetailBottomView {
        return mChatDetailActivityBottomMenu!!
    }

    fun setChatDetailActivityBottomMenu(chatDetailActivityBottomMenu: ChatDetailBottomView): ChatDetailViewOption {
        this.mChatDetailActivityBottomMenu = chatDetailActivityBottomMenu
        return this
    }
}
